docs/sphinx/source/modelchain.rst

```diff
@@ -0,0 +1,454 @@
+                      module_parameters=sandia_module,
+                      inverter_parameters=cec_inverter)
+    mc = ModelChain(system, location)
+
+Printing a ModelChain object will display its models.
+
+.. ipython:: python
+
+    print(mc)
+
+Next, we run a model with some simple weather data.
+
+.. ipython:: python
+
+    weather = pd.DataFrame([[1050, 1000, 100, 30, 5]],
+                           columns=['ghi', 'dni', 'dhi', 'temp_air', 'wind_speed'],
+                           index=[pd.Timestamp('20170401 1200', tz='US/Arizona')])
+
+    mc.run_model(times=weather.index, weather=weather);
+
+ModelChain stores the modeling results on a series of attributes. A few
+examples are shown below.
+
+.. ipython:: python
+
+    mc.aoi
+
+.. ipython:: python
+
+    mc.dc
+
+.. ipython:: python
+
+    mc.ac
+
+The remainder of this guide examines the ModelChain functionality and
+explores common pitfalls.
+
+Defining a ModelChain
+---------------------
+
+A :py:class:`~pvlib.modelchain.ModelChain` object is defined by:
+
+1. The properties of its :py:class:`~pvlib.pvsystem.PVSystem`
+   and :py:class:`~pvlib.location.Location` objects
+2. The keyword arguments passed to it at construction
+
+ModelChain uses the keyword arguments passed to it to determine the
+models for the simulation. The documentation describes the allowed
+values for each keyword argument. If a keyword argument is not supplied,
+ModelChain will attempt to infer the correct set of models by inspecting
+the Location and PVSystem attributes.
+
+Below, we show some examples of how to define a ModelChain.
+
+Let’s make the most basic Location and PVSystem objects and build from
+there.
+
+.. ipython:: python
+
+    location = Location(32.2, -110.9)
+    poorly_specified_system = PVSystem()
+    print(location)
+    print(poorly_specified_system)
+
+These basic objects do not have enough information for ModelChain to be
+able to automatically determine its set of models, so the ModelChain
+will throw an error when we try to create it.
+
+.. ipython:: python
+    :okexcept:
+
+    ModelChain(poorly_specified_system, location)
+
+If our goal is simply to get the object constructed, we can specify the
+models that the ModelChain should use. We’ll have to fill in missing
+data on the PVSystem object later, but maybe that’s desirable in some
+workflows.
+
+.. ipython:: python
+    :okexcept:
+
+    mc = ModelChain(poorly_specified_system, location,
+                    dc_model='singlediode', ac_model='snlinverter',
+                    aoi_model='physical', spectral_model='no_loss')
+    print(mc)
+
+As expected, without additional information, the
+:py:meth:`~.ModelChain.run_model` method fails at run time.
+
+.. ipython:: python
+    :okexcept:
+
+    mc.run_model(times=weather.index, weather=weather)
+
+The ModelChain attempted to execute the PVSystem object's
+:py:meth:`~pvlib.pvsystem.PVSystem.singlediode` method, and the method
```

Comment:
if reworking this section, replace ``singlediode`` with a different method

Reply:
`cec` should be our default example

Reply:
I decided to remove this section that referenced ``singlediode`` because #555 disallowed this "feature". 

The remaining examples demonstrate ``sapm`` and ``pvwatts``. I'm ok adding another example that demonstrates the behavior when pulling from the ``cec`` database if you don't think it would be pedantic. Not sure about ``pvsyst``/``desoto``.

Reply:
OK with me to remove it.  The scope of this PR is to rehost the example, so we can push off expanding the example to a follow on.